plugin-server/src/worker/ingestion/event-pipeline/cookielessServerHashStep.ts

```diff
@@ -0,0 +1,585 @@
+    // add a prefix so that we can recognise one of these in the wild
+    return 'cookieless_' + uint32ArrayToBase64String(hash).replace(/=+$/, '')
+}
+
+export function getRedisIdentifiesKey(hash: Uint32Array, teamId: number): string {
+    // assuming 6 digits for team id, this is 6 + 2 + 6 + 24 = 38 characters
+    // cklsi = cookieless identifies
+    return `cklsi:${teamId}:${uint32ArrayToBase64String(hash)}`
+}
+
+export function getRedisSessionsKey(hash: Uint32Array, teamId: number): string {
+    // assuming 6 digits for team id, this is 6 + 2 + 6 + 24 = 38 characters
+    // cklss = cookieless sessions
+    return `cklss:${teamId}:${uint32ArrayToBase64String(hash)}`
+}
+
+export function toYYYYMMDDInTimezoneSafe(
+    timestamp: number,
+    eventTimeZone: string | undefined,
+    teamTimeZone: string
+): string {
+    let dateObj: { year: number; month: number; day: number }
+    if (eventTimeZone) {
+        try {
+            dateObj = toYearMonthDayInTimezone(timestamp, eventTimeZone)
+        } catch {
+            // pass
+        }
+    }
+    try {
+        dateObj = toYearMonthDayInTimezone(timestamp, teamTimeZone)
+    } catch {
+        dateObj = toYearMonthDayInTimezone(timestamp, TIMEZONE_FALLBACK)
+    }
+    return `${dateObj.year}-${dateObj.month.toString().padStart(2, '0')}-${dateObj.day.toString().padStart(2, '0')}`
+}
+
+export function toStartOfDayInTimezoneSafe(
+    timestamp: number,
+    eventTimeZone: string | undefined,
+    teamTimeZone: string
+): Date {
+    if (eventTimeZone) {
+        try {
+            return toStartOfDayInTimezone(timestamp, eventTimeZone)
+        } catch {
+            // pass
+        }
+    }
+    try {
+        return toStartOfDayInTimezone(timestamp, teamTimeZone)
+    } catch {
+        return toStartOfDayInTimezone(timestamp, TIMEZONE_FALLBACK)
+    }
+}
+
+export function createStatelessSessionId(
+    timestamp: number,
+    eventTimezone: string | undefined,
+    teamTimeZone: string,
+    hash: Uint32Array
+): UUID7 {
+    // A sessionId is a UUIDv7, which has a timestamp part and a random part. We need to find a deterministic way to
+    // generate this ID whilst meeting the requirements of posthog session IDs
+    // see https://posthog.com/docs/data/sessions#custom-session-ids
+
+    // For the timestamp part, use the start of the day, in this user's timezone
+    const timestampOfStartOfDay = toStartOfDayInTimezoneSafe(timestamp, eventTimezone, teamTimeZone).getTime()
+
+    // For the random part, use the first 10 bytes of the hash (74 bits are actually used), as a way of ensuring
+    // determinism with no state
+    const fakeRandomBytes = Buffer.from(hash.buffer).subarray(0, 10)
+
+    return new UUID7(timestampOfStartOfDay, fakeRandomBytes)
+}
+
+export function stripPIIProperties(event: PluginEvent) {
+    if (event.properties) {
+        // we use these properties in the hash, but they should not be written to disk if explicit consent was not given
```

Comment:
> but they should not be written to disk if explicit consent was not given

You're calling this unconditionally above, so we'll assume that "if cookieless mode, then no explicit consent"?

Reply:
Yeah for now - I'm going to change this behaviour later so that cookieless events + cookie consent lets you keep these.